cpp/ql/lib/semmle/code/cpp/dataflow/internal/DataFlowImpl.qll

```diff
@@ -3313,9 +3313,14 @@ abstract private class PathNodeImpl extends PathNode {
 }
 
 /** Holds if `n` can reach a sink. */
-private predicate reach(PathNode n) { n instanceof PathNodeSink or reach(n.getASuccessor()) }
+private predicate directReach(PathNode n) {
+  n instanceof PathNodeSink or directReach(n.getASuccessor())
+}
+
+/** Holds if `n` can reach a sink or is used in a subpath. */
+private predicate reach(PathNode n) { directReach(n) or Subpaths::retReach(n) }
 
-/** Holds if `n1.getSucc() = n2` and `n2` can reach a sink. */
+/** Holds if `n1.getSucc() = n2` and `n2` can reach a sink or is used in a subpath. */
 private predicate pathSucc(PathNode n1, PathNode n2) { n1.getASuccessor() = n2 and reach(n2) }
```

Comment:
I think we should change `reach(n2)` to `directReach(n2)`, and then inline `n1.getASuccessor() = n2 and reach(n2)` in the definition of `edges` instead, since there is no need to compute `fastTC` below for subpaths.

Reply:
As agreed offline, we'll take this in a follow-up PR.

Reply:
Will fix in a follow-up PR.